app/frontend/src/components/ImageInput/ImageInput.test.tsx

```diff
@@ -232,3 +242,95 @@ describe.each`
     ).toMatch(/base64/);
   });
 });
+
+describe("ImageInput http error tests", () => {
+  beforeAll(() => {
+    server.listen();
+  });
+  beforeEach(() => {
+    const View = () => {
+      const { control } = useForm();
+      return (
+        <ImageInput
+          control={control}
+          id="image-input"
+          initialPreviewSrc={MOCK_INITIAL_SRC}
+          name="imageInput"
+          userName={NAME}
+          type="avatar"
+        />
+      );
+    };
+    render(<View />, { wrapper });
+    const uploadFile = jest.requireActual("service").service.api.uploadFile;
+    uploadFileMock.mockImplementation(uploadFile);
+    const initiateMediaUploadMock = jest.spyOn(
+      client.api,
+      "initiateMediaUpload"
+    );
+    initiateMediaUploadMock.mockResolvedValue({
+      getUploadUrl: () => "https://example.com/upload",
+    } as InitiateMediaUploadRes);
+    mockConsoleError();
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  afterAll(() => {
+    server.close();
+  });
+
+  it("displays the right error if the file is too large", async () => {
+    server.use(
+      rest.post("https://example.com/upload", async (_req, res, ctx) => {
+        return res(ctx.status(413), ctx.text("Payload too large"));
+      })
+    );
+
+    userEvent.upload(
+      screen.getByLabelText(SELECT_AN_IMAGE) as HTMLInputElement,
+      MOCK_FILE
+    );
+
+    expect(await screen.findByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+    userEvent.click(screen.getByLabelText(CONFIRM_UPLOAD));
+
+    await assertErrorAlert(IMAGE_TOO_LARGE);
+  });
+
+  it("displays a general error for server errors", async () => {
+    server.use(
+      rest.post("https://example.com/upload", async (_req, res, ctx) => {
+        return res(ctx.status(500), ctx.text("Internal server error"));
+      })
+    );
+
+    userEvent.upload(
+      screen.getByLabelText(SELECT_AN_IMAGE) as HTMLInputElement,
+      MOCK_FILE
+    );
+
+    expect(await screen.findByLabelText(CONFIRM_UPLOAD)).toBeVisible();
+    userEvent.click(screen.getByLabelText(CONFIRM_UPLOAD));
+
+    await assertErrorAlert(SERVER_ERROR);
+  });
+
+  it("displays an internal error for bad json", async () => {
+    server.use(
+      rest.post("https://example.com/upload", async (_req, res, ctx) => {
+        return res(ctx.status(200), ctx.text("[{bad: 'json'}}]"));
```

Comment:
Is this actually possible from the API server at the moment? 😱

Reply:
In theory no, but I was catching the error so thought I may as well